### DISCUSSION TITLE
stack depth limit exceeded

### DISCUSSION THREAD

**Discussion by cliveportman** [supabase/supabase#8332] [General]
```code: "54001"
details: null
hint: "Increase the configuration parameter \"max_stack_depth\" (currently 2048kB), after ensuring the platform's stack depth limit is adequate."
message: "stack depth limit exceeded"
```

Any ideas? All I'm running is:
```const {data, error} = await supabase.from('schools').insert([{ title: values.title }])```

Where values.title is nothing more than a short string.

Updates: 

1. The query runs fine with the Supabase admin panel. I've created a function doing the same thing, That works in the admin panel. I try running the function on my client? Same error.
2. The query runs fine if I turn off row-level security.

**Comment by GaryAustin1** [comment]
Seems like you may have a recursive call in your RLS policy, especially since it works when you have it off or use the sql editor.

**Reply by cliveportman** [reply]
I have:
```INSERT / Enable insert for academy users only / is_academy(uid())```

Where is_academy() is: 
```create or replace function is_academy(user_id uuid) 
returns boolean
language sql
as $$
  select exists (select * FROM academy WHERE academy.id = user_id)
$$;
```

And it tests positive with the current user I'm trying to insert with. The I have a standard select with all authenticated users. Is there a way to view all functions linked to a table or object?

**Reply by GaryAustin1** [reply]
Does academy table have RLS on it?  (assuming that is_academy() is not security definer type, it will have to meet that also) 
Any chance that RLS links back to schools table?
Normally you use a security definer function if you need to avoid getting trapped in recursive RLS policies.

Easy way to see triggers is just to use the dashboard, database, triggers UI (if you are on a SB instance).  Otherwise there are queries you could google to find constraints or triggers on tables.
